src/rusty/src/rest_downloader.rs

```diff
@@ -0,0 +1,325 @@
+                                    bytes_skipped += 1;
+                                    if actual_len + bytes_skipped > bytes_read {
+                                        // Go back and get more bytes so we can skip trailing \n
+                                        chunk_remaining = 0;
+                                    }
+                                },
+                                Some(_) => {},
+                                None => {
+                                    // Go back and get more bytes so we can skip trailing \n
+                                    chunk_remaining = 0;
+                                },
+                            }
+                            break;
+                        },
+                        Err(_) => return None,
+                    }
+                }
+                if chunk_remaining != 0 {
+                    bytes_read -= bytes_skipped;
+                    resp.drain(actual_len..actual_len + bytes_skipped);
+                    if actual_len + chunk_remaining > max_resp { return None; }
+                    let already_in_chunk = cmp::min(bytes_read - actual_len, chunk_remaining);
+                    actual_len += already_in_chunk;
+                    chunk_remaining -= already_in_chunk;
+                    continue 'read_bytes;
+                } else {
+                    if finished_read {
+                        // Note that we may leave some extra \r\ns to be read, but that's OK,
+                        // we'll ignore then when parsing headers for the next request.
+                        resp.resize(actual_len, 0);
+                        return Some(resp);
+                    } else {
+                        // Need to read more bytes to figure out chunk length
+                    }
+                }
+            }
+            resp.resize(bytes_read + cmp::max(10, chunk_remaining), 0);
+            let avail = read_socket!();
+            bytes_read += avail;
+            if chunk_remaining != 0 {
+                let chunk_read = cmp::min(chunk_remaining, avail);
+                chunk_remaining -= chunk_read;
+                actual_len += chunk_read;
+            }
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn init_fetch_rest_blocks(uri: *const u8) -> bool {
+    let mut uri_vec = Vec::with_capacity(50);
+    for i in 0..10000 {
+        let c = unsafe { *uri.offset(i) };
+        if c == 0 { break; }
+        uri_vec.push(c);
+    }
+    let uri_str = match String::from_utf8(uri_vec) {
+        Ok(r) => r,
+        Err(_) => return false,
+    };
+    // Sadly only non-SSL is supported for now
+    if let Some((false, _, _, _)) = split_uri(&uri_str) { } else { return false; }
+    std::thread::spawn(move || {
+        // Always catch panics so that even if we have some bug in our parser we don't take the
+        // rest of Bitcoin Core down with us:
+        let _ = catch_unwind(move || {
```

Comment:
Could we easily log a panic easily somehow? (although this introduces a lot of complications to pass the `datadir` location somehow, could we maybe use some tmp location just as a jump start?)

Reply:
Right, it would be really nice to be able to call LogPrintf from rust code, but I'd prefer to leave that to a future PR.